core-web/libs/template-builder/src/lib/components/template-builder/template-builder.component.spec.ts

```diff
@@ -27,25 +41,70 @@ describe('TemplateBuilderComponent', () => {
             }
         ]
     });
-
     beforeEach(() => {
-        spectator = createComponent({
-            props: {
-                templateLayout: {
-                    body: FULL_DATA_MOCK,
-                    footer: false,
-                    header: false,
-                    sidebar: {},
-                    title: '',
-                    width: ''
+        spectator = createHost(
+            `<dotcms-template-builder [templateLayout]="templateLayout"></dotcms-template-builder>`,
+            {
+                hostProps: {
+                    templateLayout: { body: FULL_DATA_MOCK }
                 }
             }
+        );
+
+        store = spectator.inject(DotTemplateBuilderStore);
+    });
+
+    it('should create', () => {
+        expect(spectator).toBeTruthy();
+    });
+
+    it('should have a Add Row Button', () => {
+        expect(spectator.query(byTestId('add-row'))).toBeTruthy();
+    });
+
+    it('should have a Add Box Button', () => {
+        expect(spectator.query(byTestId('add-box'))).toBeTruthy();
+    });
+
+    it('should have a 12 columns as background', () => {
+        expect(spectator.queryAll(byTestId('column')).length).toBe(12);
```

Comment:
This should be inside the component that draw this, here we just need to look for that component.